battleShip.py

```diff
@@ -0,0 +1,283 @@
+                    break
+
+            except ValueError:
+                print("Invalid coordinates. Try again :<")
+
+    def is_ship_destroyed(self, ship, grid): # Check if the ship is destroyed
+        for i in range(len(grid)):
+            for j in range(len(grid)):
+                if grid[i][j] == 'S ' and (i, j) not in self.destroyedShips:
+                    return False
+        return True
+
+    def display_grid(self): # Display the grid
+        size = len(self.grid)
+
+        # Display column labels
+        print("   ", end="")
+        for i in range(size):
+            print(f"{chr(i + ord('A')):2}", end=" ")
+        print()
+
+        # Display row labels and grid
+        for i in range(size):
+            print(f"{i+1:2}", end=" ")
+            print(" ".join(self.grid[i]))
+
+    def display_hidden_grid(self,opponent):  # Display the hidden grid
+        size = len(self.hiddenGrid)
+
+        # Display column labels
+        print("   ", end="")
+        for i in range(size):
+            print(f"{chr(i + ord('A')):2}", end=" ")
+        print()
+
+        # Display row labels and grid
+        for i in range(size):
+            print(f"{i + 1:2}", end=" ")
+            for j in range(size):
+                if self.hiddenGrid[i][j] == 'S ' or self.hiddenGrid[i][j] == 'M ' or self.hiddenGrid[i][j] == 'X ':
+                    print(self.hiddenGrid[i][j], end=" ")
+                else:
+                    print(self.hiddenGrid[i][j], end=" ")
+            print()
+
+    def save_game(self, obj2):  # Save the game
+        with open("savegame.pickle", "wb") as file:
+            pickle.dump(obj2, file)
+            pickle.dump(self, file)
+        print("\t\tGame saved successfully :>)")
+
+
+def display_menu():
+    print("\t\t----== Battle Ships Menu ----==")
+    print("\t\t1. New Game")
+    print("\t\t2. Load Game")
+    print("\t\t3. Quit")
+
+
+def display_menu2():
+    print("\t\t 1. Play with a friend")
+    print("\t\t 2. Play with a computer")
+    print("\t\t 3. Back to main menu")
+
+def load_game(): # Load the game
+    try:
+        list=[]
+        with open("savegame.pickle", "rb") as file:
+            p1=pickle.load(file)
+            list.append(p1)
+            p2=pickle.load(file)
+            list.append(p2)
+            return list
+    except FileNotFoundError:
+        print("No saved game found :<(")
+        return None
+def main():
+    while True:
+        display_menu()
+        choice = input("\t\tEnter your choice: ")
+
+        if choice == '1':
+            while True:
+                display_menu2()
+                choice = input("\t\tEnter your choice: ")
+                if choice == '1':
+                    p = input("\t\tPlayer 1 Enter your name :) ")
+                    p2 = input("\t\tPlayer 2 Enter your name :) ")
+                    player1 = Player(p)
+                    player2 = Player(p2)
+
+                    player1.create_grid(26)
```

Comment:
still creating two grids, also as a human player i attacked ```A1``` but when i got turn again the tile was still marked ```-``` it should be something like ```M (for missed) and X (for ship tile)```